slapo/op/attention.py

```diff
@@ -436,27 +444,24 @@ def reshape_for_scores(self, x: torch.Tensor):
     def forward(
         self,
         hidden_states: Optional[tuple[torch.FloatTensor]],
-        layer_past: Optional[tuple[torch.Tensor]] = None,
         attention_mask: Optional[torch.FloatTensor] = None,
+        layer_past: Optional[tuple[torch.Tensor]] = None,
```

Comment:
Why change the argument order? In this case you also need to change gpt2/schedule.py accordingly.

Reply:
I think it would be better to align the signature of the BertAttention. Since layer_past can be sometimes set as `None`, it may easily cause problem for module tracing and pipeline data transmission. TorchScript cannot trace this kind of module just because of the argument order.

Reply:
I'll change the GPT2 schedule accordingly.

Reply:
Ok fair enough.